lib/sort_by_length.rb

```diff
@@ -1,7 +1,35 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) insertion sort
+# Space complexity: 0(n) created new array&hash that have the linear relationship with the length of the input
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  my_sentence = my_sentence.split(" ")
+  sentence_hash = Hash.new
+  my_sentence.map do |word|
+    sentence_hash[word] = word.length
+  end
```

Comment:
Why use `.map` if you are going to add each element to the hash like you do.  I think this is better as a `.each` loop.